core/services/workflows/engine.go

```diff
@@ -111,6 +111,7 @@ type Engine struct {
 	stopCh               services.StopChan
 	newWorkerTimeout     time.Duration
 	maxExecutionDuration time.Duration
+	heartbeatCadence     time.Duration
```

Comment:
/nit `heartbeat` is likely fine since its type is a time.Duration

Reply:
one potential source of conflict in the future re:naming it heartbeat, is that my understanding is that this is a user facing heartbeat. We may desire to have an internal heartbeat in the future.

Reply:
gotcha. what would you recommend the name could be, to avoid that in the future?

Reply:
`externalHeartbeat` maybe? @jmank88 may have thoughts